options.go

```diff
@@ -55,6 +55,39 @@ type options struct {
 	gCHeapOpts *typeOption
 	cpuOpts    *typeOption
 	threadOpts *typeOption
+
+	pReportOpts *ReporterOptions
+}
+
+// rptEvent stands of the args of report event
+type rptEvent struct {
+	PType   string
+	Buf     []byte
+	Reason  string
+	EventID string
+}
+
+type ReporterOptions struct {
+	L        *sync.RWMutex
```

Comment:
I don't think we need another lock here. just reuse the options.L is good enough.